app/javascript/authentication/SignInForm.tsx

```diff
@@ -10,40 +10,76 @@ import {
   AlertBox,
   SiteAlert,
   LinkButton,
+  debounce,
 } from "@bloom-housing/ui-components"
-import { useForm } from "react-hook-form"
+import { Dialog, Link, Heading, Alert } from "@bloom-housing/ui-seeds"
+import { FieldValues, SubmitHandler, useForm } from "react-hook-form"
 
 import { getMyAccountPath } from "../util/routeUtil"
-import UserContext from "./context/UserContext"
 import EmailFieldset from "../pages/account/components/EmailFieldset"
 import PasswordFieldset from "../pages/account/components/PasswordFieldset"
-import { Link, Heading } from "@bloom-housing/ui-seeds"
 import "../pages/account/styles/account.scss"
+import { AxiosError } from "axios"
+import { confirmEmail, signIn } from "../api/authApiService"
 
-const SignInForm = () => {
-  const { signIn } = useContext(UserContext)
+const NotConfirmedModal: React.FC<{
```

Comment:
suggestion: name this AccountNotConfirmedModal for specificity?

suggestion: Our pattern for react components has been to do 
```
const NotConfirmedModal = (  {  email, onClose }: {  email: string, onClose: () => void } ) => {
    return ()
}
```

Can we try to stick with that?

Reply:
question: In this and a previous PR, I see that you are starting to use `React.FunctionComponent`. This seems to be discouraged [elsewhere](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/), but I would love to know the reasoning behind this usage.